--- config/ltmain.sh
+++ config/ltmain.sh
@@ -6417,8 +6417,15 @@
 	func_source "$lib"
 
 	# Convert "-framework foo" to "foo.ltframework"
+        # and " -pthread" to " -Wl,-pthread" if NAG compiler
 	if test -n "$inherited_linker_flags"; then
-	  tmp_inherited_linker_flags=`$ECHO "$inherited_linker_flags" | $SED 's/-framework \([^ $]*\)/\1.ltframework/g'`
+          func_basename "$CC"
+          case "$func_basename_result" in
+            *nagfor*)
+	      tmp_inherited_linker_flags=`$ECHO " $inherited_linker_flags" | $SED 's/-framework \([^ $]*\)/\1.ltframework/g; s/ -pthread/ -Wl,-pthread/g'`;;
+            *)
+	      tmp_inherited_linker_flags=`$ECHO "$inherited_linker_flags" | $SED 's/-framework \([^ $]*\)/\1.ltframework/g'`;;
+          esac
 	  for tmp_inherited_linker_flag in $tmp_inherited_linker_flags; do
 	    case " $new_inherited_linker_flags " in
 	      *" $tmp_inherited_linker_flag "*) ;;
@@ -8001,6 +8008,18 @@
 	  ;;
       esac
 
+      # Time to revert changes made for NAG. We have experienced problems
+      # only with new_inherited_linker_flags so far, so we change only it.
+      # We do realize that we might "revert" a change that we have not
+      # actually introduced (i.e. we might be changing " -Wl,-pthread" that
+      # we have received in this very form) but since it is very unlikely,
+      # we decided to do so for the matter of simplicity.
+      func_basename "$CC"
+      case "$func_basename_result" in
+        *nagfor*)
+          new_inherited_linker_flags=`$ECHO " $new_inherited_linker_flags" | $SED 's% -Wl,-pthread% -pthread%g'`;;
+      esac
+
       # move library search paths that coincide with paths to not yet
       # installed libraries to the beginning of the library search list
       new_libs=
